### PR TITLE
[crypto][kimchi-backend] removing unecessary r1cs offset trick

### PR DESCRIPTION
**Pull request by mimoo**
removes the offset in inputs in snarky, this was because of R1CS which we don't use anymore (see https://github.com/o1-labs/snarky/pull/658)

### CODE REVIEW

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

File: src/lib/crypto/snarky_tests/examples/output.json [hunk 1]
```diff
@@ -1 +1 @@
...
...,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]},{"typ":"Generic","wires":[{"row":2,"col":0},{"row":2,"col":1},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":2,"col":5},{"row":2,"col":6}],"coeffs":[[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]},{"typ":"Generic","wires":[{"row":0,"col":0},{"row":1,"col":0},{"row":2,"col":2},{"row":2,"col":3},{"row":2,"col":4},{"row":2,"col":5},{"row":2,"col":6}],"coeffs":[[0,0,0,0,237,48,45,153,27,249,76,9,252,152,70,34,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,64],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]}]}
```
Comment (mimoo) [inline src/lib/crypto/snarky_tests/examples/output.json:1]: first:

```
row0.pub.Generic<1,0,0,0,0>
.l1 -> row2.r1

row1.pub.Generic<1,0,0,0,0>
.l1 -> row2.l1

row2.Generic<-1,0,0,1,0>
.l1 -> row0.l1, .r1 -> row1.l1
```

second can't deserialize correctly due to a `col: 6` in the serialization... mmm...

**Comment by mimoo** [comment]
wait... the serialization shouldn't change due to the removal of that r1cs row o.o

File: src/lib/crypto/snarky_tests/snarky_tests.ml [hunk 2]
```diff
@@ -325,21 +328,30 @@ module As_prover_circuits = struct
...
+            List.fold l ~init:Impl.Field.Constant.zero ~f
           in
-          assert (not (Impl.Field.(Constant.compare total Constant.one) = 0)) ;
-          assert (not Impl.Field.Constant.(equal total one)) ;
-          () ) ;
+
+          (* manual sum is equal to circuit sum *)
+          let abc = Impl.As_prover.read_var abc in
+
+          assert (Impl.Field.(Constant.equal abc total)) ) ;
```
Comment (mimoo) [inline src/lib/crypto/snarky_tests/snarky_tests.ml:354]: @fabrizio-m I slightly changed this test to check that I can read the sum `abc` and check that it's equal to the indexed variables. Now that I'm looking into this test I'm wondering why we're really testing these, as nobody should access variables manually like this (via `Unsafe.of_index`). Maybe we can delete this test in the future?

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

File: src/lib/crypto/snarky_tests/snarky_tests.ml (hunk 2 above)
Comment (fabrizio-m) [reply src/lib/crypto/snarky_tests/snarky_tests.ml:354]: I think the idea was to check that the indexing was consistent through both implementations.

**Review by fabrizio-m — APPROVED** [review]

File: src/lib/crypto/snarky_tests/snarky_tests.ml (hunk 2 above)
Comment (mimoo) [reply src/lib/crypto/snarky_tests/snarky_tests.ml:354]: yeah that makes sense : o I've kept that behavior so it'll still get tested!

**Comment by mimoo** [comment]
ping @mrmr1993 can you review this one as well? I can't merge it without a protocol approval : o

**Review by mrmr1993 — APPROVED** [review]

**Comment by mimoo** [comment]
!ci-build-me

**Pull request merged**